libdnf/rpm/solv_query.cpp

```diff
@@ -1351,4 +1728,78 @@ std::size_t SolvQuery::size() const noexcept {
     return p_impl->query_result.size();
 }
 
+std::pair<bool, libdnf::rpm::Nevra> SolvQuery::resolve_pkg_spec(
+    const std::string & pkg_spec,
+    bool icase,
+    bool with_nevra,
+    bool with_provides,
+    bool with_filenames,
+    bool with_src,
+    const std::vector<libdnf::rpm::Nevra::Form> & forms) {
+    SolvSack * sack = p_impl->sack.get();
+    Pool * pool = sack->pImpl->get_pool();
+    solv::SolvMap filter_result(static_cast<int>(sack->pImpl->get_nsolvables()));
+    if (with_nevra) {
+        const std::vector<Nevra::Form> & test_forms = forms.empty() ? Nevra::PKG_SPEC_FORMS : forms;
+        Nevra nevra_obj;
+        for (auto form : test_forms) {
+            if (nevra_obj.parse(pkg_spec, form)) {
+                p_impl->filter_nevra(
+                    nevra_obj,
+                    true,
+                    icase ? libdnf::sack::QueryCmp::IGLOB : libdnf::sack::QueryCmp::GLOB,
+                    filter_result,
+                    with_src);
+                filter_result &= p_impl->query_result;
+                if (!filter_result.empty()) {
+                    // Apply filter results to query
+                    p_impl->query_result &= filter_result;
+                    return {true, libdnf::rpm::Nevra(std::move(nevra_obj))};
+                }
+            }
+        }
+        if (forms.empty()) {
+            auto & sorted_solvables = p_impl->sack->pImpl->get_sorted_solvables();
+            p_impl->filter_nevra(
+                pool,
+                sorted_solvables,
+                pkg_spec,
+                true,
+                icase ? libdnf::sack::QueryCmp::IGLOB : libdnf::sack::QueryCmp::GLOB,
+                filter_result);
+            filter_result &= p_impl->query_result;
+            if (!filter_result.empty()) {
+                p_impl->query_result &= filter_result;
+                return {true, libdnf::rpm::Nevra()};
+            }
+        }
+    }
+    if (with_provides) {
+        ReldepList reldep_list(sack);
+        str2reldep_internal(reldep_list, libdnf::sack::QueryCmp::GLOB, true, pkg_spec);
+        sack->pImpl->make_provides_ready();
+        p_impl->filter_provides(pool, libdnf::sack::QueryCmp::EQ, reldep_list, filter_result);
+        filter_result &= p_impl->query_result;
+        if (!filter_result.empty()) {
+            p_impl->query_result &= filter_result;
+            return {true, libdnf::rpm::Nevra()};
+        }
+    }
+    if (with_filenames && is_file_pattern(pkg_spec)) {
+        filter_dataiterator(
+            pool,
+            SOLVABLE_FILELIST,
+            SEARCH_FILES | SEARCH_COMPLETE_FILELIST | SEARCH_GLOB,
```

Comment:
And what about ` | (icase ? SEARCH_NOCASE : 0)`.
Oh original DNF code also ignores `icase` argument. It seems as bug in the old DNF.

Reply:
Is it a bug? Probably not, because the method with ICASE is used for hints where you run the same argument with ICASE. Then you provide hint with package a name.
No, you all right, this is bug but also incorrect usage from DNF. API must be general and then particular setting must be applied by the way that allows specific user case.